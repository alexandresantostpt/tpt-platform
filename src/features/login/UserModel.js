import { config } from '@config'
import { roles } from '@/constants/roles'

class UserModel {
    constructor({ __v, _id, active, agency, blocked, cellPhone, cpf, email, image, logo, name, password, role, theme }) {
        this.__v = __v
        this._id = _id
        this._active = active
        this._agency = agency
        this._blocked = blocked
        this._cellPhone = cellPhone
        this._cpf = cpf
        this._email = email
        this._image = image
        this._logo = logo
        this._name = name
        this._password = password
        this._role = role
        this._theme = theme
    }

    get v() {
        return this.__v
    }

    get active() {
        return this._active
    }

    get agency() {
        return this._agency
    }

    get blocked() {
        return this._blocked
    }

    get cellPhone() {
        return this._cellPhone
    }

    get cpf() {
        return this._cpf
    }

    get email() {
        return this._email
    }

    get id() {
        return this._id
    }

    get image() {
        return this._image
    }

    getImage() {
        return this.image && `${config.api.url}/user/download/${this.image}`
    }

    get logo() {
        return this._logo
    }

    get name() {
        return this._name
    }

    get password() {
        return this._password
    }

    get role() {
        return this._role
    }

    get theme() {
        return this._theme
    }

    isAdmin = () => this._role === roles.ADMIN

    isConsultant = () => this._role === roles.CONSULTANT

    isMaster = () => this._role === roles.MASTER
}

export default UserModel
