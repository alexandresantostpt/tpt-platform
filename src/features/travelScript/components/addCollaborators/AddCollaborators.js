import React, { useState } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { i18n } from '@i18n'

import InviteButton from '../inviteButton/InviteButton'
import Input from '../input/Input'
import { isNotEmpty } from '@/utils/functions'
import { keys } from '@/constants/keys'
import Avatars from './Avatars'
import UserModel from '@/features/login/UserModel'
import If from '@/components/if/If'

const AddCollaborators = ({ addCollaborator, collaborators, removeCollaborator }) => {
    const [shouldShowInput, updateShouldShowInput] = useState(false)
    const [email, updateEmail] = useState('')

    const AddCollaboratorsContainer = styled.div`
        align-items: flex-end;
        display: flex;
        flex-direction: ${isNotEmpty(collaborators) && shouldShowInput ? 'column' : 'row'};
    `

    const handleKeyPress = event => {
        if (isNotEmpty(email) && event.key === keys.ENTER) {
            updateShouldShowInput(false)
            addCollaborator(email)
            updateEmail('')
        }
    }

    const avatarClick = collaborator => {
        updateShouldShowInput(true)
        updateEmail(collaborator.email)
    }

    /* eslint-disable jsx-a11y/no-autofocus */
    return (
        <AddCollaboratorsContainer>
            <If condition={isNotEmpty(collaborators)}>
                <Avatars collaborators={collaborators} onClick={avatarClick} onRemove={removeCollaborator} />
            </If>
            <If
                condition={shouldShowInput}
                el={<InviteButton onClick={() => updateShouldShowInput(true)} round={isNotEmpty(collaborators)} />}
            >
                <Input
                    autoFocus
                    onBlur={() => updateShouldShowInput(false)}
                    onChange={({ target }) => updateEmail(target.value)}
                    onKeyPress={handleKeyPress}
                    placeholder={i18n.t('placeholders.shareTravelScript')}
                    value={email}
                />
            </If>
        </AddCollaboratorsContainer>
    )
}

AddCollaborators.defaultProps = {
    collaborators: []
}

AddCollaborators.propTypes = {
    addCollaborator: PropTypes.func.isRequired,
    collaborators: PropTypes.arrayOf(UserModel),
    removeCollaborator: PropTypes.func.isRequired
}

export default AddCollaborators
