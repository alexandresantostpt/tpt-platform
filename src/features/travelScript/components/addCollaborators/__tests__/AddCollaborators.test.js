import React from 'react'
import renderer from 'react-test-renderer'

import AddCollaborators from '../AddCollaborators'
import UserModel from '@/features/login/UserModel'

describe('Tests for AddCollaborator component', () => {
    it('Should render without crash', () => {
        const tree = renderer.create(<AddCollaborators addCollaborator={null} collaborators={[]} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('Should render with given props', () => {
        const tree = renderer
            .create(
                <AddCollaborators
                    addCollaborator={null}
                    collaborators={[
                        new UserModel({ email: 'test1@email.com', name: 'test1' }),
                        new UserModel({ email: 'test2@email.com', name: 'test2' })
                    ]}
                />
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
