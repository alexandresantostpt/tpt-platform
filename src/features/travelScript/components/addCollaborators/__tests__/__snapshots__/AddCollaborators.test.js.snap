// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for AddCollaborator component Should render with given props 1`] = `
<div
  className="AddCollaborators__AddCollaboratorsContainer-sc-9j6qbx-0 jhjoiw"
>
  <div
    className="Avatars__AvatarsContainer-b9jyqp-2 kWDNTJ"
  >
    <div
      className="Avatars__AvaatarsWrap-b9jyqp-3 icEjmW"
    >
      <div
        className="DefaultAvatar__AvatarContainer-sc-1lk30og-0 dWocpW"
        data-tip="test1"
        onClick={[Function]}
        size={2.3125}
      >
        T
      </div>
      <div
        className="__react_component_tooltip place-top type-dark "
        data-id="tooltip"
      />
      <button
        className="Avatars__AvatarsButton-b9jyqp-0 hQNgWs"
        onClick={[Function]}
      >
        <span
          className="Avatars__AvatarsIcon-b9jyqp-1 gZdRTM"
        />
      </button>
    </div>
    <div
      className="Avatars__AvaatarsWrap-b9jyqp-3 icEjmW"
    >
      <div
        className="DefaultAvatar__AvatarContainer-sc-1lk30og-0 dWocpW"
        data-tip="test2"
        onClick={[Function]}
        size={2.3125}
      >
        T
      </div>
      <div
        className="__react_component_tooltip place-top type-dark "
        data-id="tooltip"
      />
      <button
        className="Avatars__AvatarsButton-b9jyqp-0 hQNgWs"
        onClick={[Function]}
      >
        <span
          className="Avatars__AvatarsIcon-b9jyqp-1 gZdRTM"
        />
      </button>
    </div>
  </div>
  <div
    className="AddButton-sc-3y6odv-0 cOrHuk"
    onClick={[Function]}
  >
    +
  </div>
</div>
`;

exports[`Tests for AddCollaborator component Should render without crash 1`] = `
<div
  className="AddCollaborators__AddCollaboratorsContainer-sc-9j6qbx-0 jhjoiw"
>
  <button
    className="StyledInviteButton-sc-1iige8l-0 cJgVMw"
    onClick={[Function]}
  >
    Convidar
  </button>
</div>
`;
